internal/dag/secret_test.go

```diff
@@ -473,24 +341,31 @@ func TestIsValidSecret(t *testing.T) {
 					CRLKey: []byte(""),
 				},
 			},
-			valid: false,
-			err:   errors.New("can't use zero-length crl.pem value"),
+			tlsSecretError: errors.New(`secret type is not "kubernetes.io/tls"`),
+			caSecretError:  errors.New(`empty "ca.crt" key`),
+			crlSecretError: errors.New(`empty "crl.pem" key`),
+		},
+		"kubernetes.io/dockercfg Secret, with TLS cert, CA cert and CRL": {
+			secret: &v1.Secret{
+				Type: v1.SecretTypeDockercfg,
+				Data: map[string][]byte{
+					v1.TLSCertKey:       []byte(fixture.CERTIFICATE),
+					v1.TLSPrivateKeyKey: []byte(fixture.RSA_PRIVATE_KEY),
+					CACertificateKey:    []byte(fixture.CA_CERT),
+					CRLKey:              []byte(fixture.CRL),
+				},
+			},
+			tlsSecretError: errors.New(`secret type is not "kubernetes.io/tls"`),
+			caSecretError:  errors.New(`secret type is not "kubernetes.io/tls" or "Opaque"`),
+			crlSecretError: errors.New(`secret type is not "kubernetes.io/tls" or "Opaque"`),
 		},
 	}
 
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			type Result struct {
-				Valid bool
-				Err   error
-			}
-
-			want := Result{Valid: tc.valid, Err: tc.err}
-
-			valid, err := isValidSecret(tc.secret)
-			got := Result{Valid: valid, Err: err}
-
-			assert.Equal(t, want, got)
+			assert.Equal(t, tc.tlsSecretError, validTLSSecret(tc.secret))
```

Comment:
nice way of testing this, since the way the Secret type is constructed it means we can have a single secret used in different ways